samples/matter/bridge/src/bridge_manager.cpp

```diff
@@ -44,20 +61,23 @@ CHIP_ERROR BridgeManager::AddBridgedDevice(BridgedDevice::DeviceType bridgedDevi
 #if CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE
 	case BridgedDevice::DeviceType::OnOffLight: {
 		LOG_INF("Adding OnOff Light bridged device");
+		dataProvider.reset(Platform::New<OnOffLightDataProvider>(BridgeManager::HandleUpdate));
 		device.reset(Platform::New<OnOffLightDevice>(nodeLabel));
 		break;
 	}
 #endif /* CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE */
 #if CONFIG_BRIDGE_TEMPERATURE_SENSOR_BRIDGED_DEVICE
 	case BridgedDevice::DeviceType::TemperatureSensor: {
 		LOG_INF("Adding TemperatureSensor bridged device");
+		dataProvider.reset(Platform::New<TemperatureSensorDataProvider>(BridgeManager::HandleUpdate));
 		device.reset(Platform::New<TemperatureSensorDevice>(nodeLabel));
 		break;
 	}
 #endif /* CONFIG_BRIDGE_TEMPERATURE_SENSOR_BRIDGED_DEVICE */
 #if CONFIG_BRIDGE_HUMIDITY_SENSOR_BRIDGED_DEVICE
 	case BridgedDevice::DeviceType::HumiditySensor: {
 		LOG_INF("Adding HumiditySensor bridged device");
+		dataProvider.reset(Platform::New<HumiditySensorDataProvider>(BridgeManager::HandleUpdate));
```

Comment:
The dataProvider is released in case everything went good (l.100). In case something went wrong or we want to remove the device, the dataProvider will be removed as well.